docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -1,632 +1,146 @@
-That's why if you want to change the public key controlling the Account, you would send a transaction targeting the very Account contract:
-
-[,python]
-----
-await signer.send_transaction(
-    account,
-    account.contract_address,
-    'set_public_key',
-    [NEW_KEY]
-)
-----
-
-Or if you want to update the Account's L1 address on the `AccountRegistry` contract, you would
-
-[,python]
-----
-await signer.send_transaction(account, registry.contract_address, 'set_L1_address', [NEW_ADDRESS])
-----
-
-NOTE: You can read more about how messages are structured and hashed in the https://github.com/OpenZeppelin/cairo-contracts/discussions/24[Account message scheme  discussion].
-For more information on the design choices and implementation of multicall, you can read the https://github.com/OpenZeppelin/cairo-contracts/discussions/27[How should Account multicall work discussion].
-
-The `\\__validate__` and `\\__execute__` methods accept the same arguments; however, `\\__execute__` returns a transaction response:
-
-[,cairo]
-----
-func __validate__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*) {
-}
+TIP: For detailed information on the usage and implementation check the xref:/api/account.adoc[API Reference] section.
 
-func __execute__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*
-) -> (response_len: felt, response: felt*) {
-}
-----
+== Table of Contents
 
-Where:
+* <<basic_account,Basic Account>>
+** <<account_entrypoints,Account entrypoints>>
+** <<interface,Interface>>
+* <<standard_account_interface,Standard Account Interface>>
+** <<isrc6_interface,ISRC6 Interface>>
+** <<isrc5_interface,ISRC5 Interface>>
+* <<deploying_an_account,Deploying an Account>>
 
-* `call_array_len` is the number of calls.
-* `call_array` is an array representing each `Call`.
-* `calldata_len` is the number of calldata parameters.
-* `calldata` is an array representing the function parameters.
+== Basic Account
 
-NOTE: The scheme of building multicall transactions within the `\\__execute__` method will change once StarkNet allows for pointers in struct arrays.
-In which case, multiple transactions can be passed to (as opposed to built within) `\\__execute__`.
+As we mentioned, Accounts in Starknet are smart contracts, and so they can be deployed and interacted
+with like any other contract, and can be extended to implement any custom logic. However, the Account is a special
+type of contract that is used to validate and execute transactions, and there are a set of entrypoints
+that must be implemented for enabling these features.
 
-There's a fourth canonical entrypoint for accounts, the `\\__validate_deploy__` method. It is **only callable by the protocol** during the execution of a `DeployAccount` type of transaction, but not by any other contract. This entrypoint is for counterfactual deployments.
+=== Account entrypoints
```

Comment:
I still have mixed feelings regarding this, because:

> (e.g. ppl thinking implementing just these methods makes up for a valid, minimal account)

They d make up for a valid, minimal account, and [the interface presented in corelib](https://github.com/starkware-libs/cairo/blob/main/corelib/src/starknet/account.cairo#L11) is still not the SRC6 one.

I understand the idea of standardization, but presenting SRC6 as the minimum valid account, still feels strange to me. We can have a minimum account, and then have SRC6, with no further theoretical issues imo.

Reply:
Agree. "basic account" should account for that -- we're not saying it's the bare minimum physically possible account, just that it makes up for a _basic_ one. Although such minimal an account is possible, I still believe our library should opinionatedly encourage and present SRC6 as the canonical way of implementing an account -- we make a library of standars.

Reply:
Updated! Lmk what you think.